lib/utils.js

```diff
@@ -0,0 +1,37 @@
+const { curry } = require('ramda');
+const { KEY_START, KEY_END, RSA_KEY_START, RSA_KEY_END } = require('./constants');
+const domainRegTest = /^[a-z0-9-]+$/;
```

Comment:
may not be super relevant where this is used, but I don't think `-` is allowed at the beginning or end of a subdomain. Just something to think about when you're using this regex to say that something is a valid subdomain or not.